src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/HentaiVNParser.kt

```diff
@@ -0,0 +1,285 @@
+import org.koitharu.kotatsu.parsers.util.attrAsRelativeUrl
+import org.koitharu.kotatsu.parsers.util.attrAsRelativeUrlOrNull
+import org.koitharu.kotatsu.parsers.util.domain
+import org.koitharu.kotatsu.parsers.util.generateUid
+import org.koitharu.kotatsu.parsers.util.insertCookies
+import org.koitharu.kotatsu.parsers.util.mapChapters
+import org.koitharu.kotatsu.parsers.util.mapNotNullToSet
+import org.koitharu.kotatsu.parsers.util.parseHtml
+import org.koitharu.kotatsu.parsers.util.requireElementById
+import org.koitharu.kotatsu.parsers.util.selectFirstOrThrow
+import org.koitharu.kotatsu.parsers.util.selectLast
+import org.koitharu.kotatsu.parsers.util.toAbsoluteUrl
+import org.koitharu.kotatsu.parsers.util.toIntUp
+import org.koitharu.kotatsu.parsers.util.tryParse
+import org.koitharu.kotatsu.parsers.util.urlBuilder
+import org.koitharu.kotatsu.parsers.util.urlEncoded
+import java.text.SimpleDateFormat
+import java.util.EnumSet
+
+private const val PAGE_SIZE = 15
+private const val SEARCH_PAGE_SIZE = 10
+
+@MangaSourceParser("HENTAIVN", "HentaiVN", "vi")
+class HentaiVNParser(context: MangaLoaderContext) : MangaParser(context, MangaSource.HENTAIVN) {
+
+	override val configKeyDomain: ConfigKey.Domain = ConfigKey.Domain("hentaivn.autos", "hentaivn.tv")
+
+	override val sortOrders: Set<SortOrder> = EnumSet.of(
+			SortOrder.UPDATED,
+			SortOrder.POPULARITY,
+			SortOrder.RATING,
+			SortOrder.NEWEST
+		)
+
+	private val tagCache = SuspendLazy(this::fetchTags)
+
+	override suspend fun getDetails(manga: Manga): Manga = coroutineScope {
+		val chapterDeferred = async { fetchChapters(manga.url) }
+		val docs = webClient.httpGet(manga.url.toAbsoluteUrl(domain)).parseHtml()
+		val infoEl = docs.selectFirstOrThrow("div.container")
+			.selectFirstOrThrow("div.page-info")
+
+		manga.copy(
+			altTitle = infoEl.infoText("Tên Khác:"),
+			author = infoEl.infoText("Tác giả"),
+			description = infoEl.selectFirst("p:contains(Nội dung:)")
+				?.nextElementSibling()
+				?.outerHtml(),
+			tags = tagCache.tryGet().getOrNull()?.let { tagMap ->
+				infoEl.selectFirst("p:contains(Thể Loại:)")
+					?.select("span > a")
+					?.mapNotNullToSet {
+						tagMap[it.text()]
+					}
+			}.orEmpty(),
+			state = infoEl.infoText("Tình Trạng:")?.let {
+				when (it) {
+					"Đã hoàn thành" -> MangaState.FINISHED
+					"Đang tiến hành" -> MangaState.ONGOING
+					else -> null
+				}
+			},
+			rating = docs.selectFirst("div.page_like")?.let {
+				val like = it.selectFirst("div.but_like")?.text()?.trim()?.toIntOrNull() ?: return@let null
+				val dislike = it.selectFirst("div.but_unlike")?.text()?.trim()?.toIntOrNull() ?: return@let null
+				when {
+					like == 0 && dislike == 0 -> RATING_UNKNOWN
+					else -> like.toFloat() / (like + dislike)
+				}
+			} ?: RATING_UNKNOWN,
+			chapters = chapterDeferred.await(),
+		)
+	}
+
+	override suspend fun getList(offset: Int, query: String?, tags: Set<MangaTag>?, sortOrder: SortOrder): List<Manga> {
```

Comment:
Consider using `PagedMangaParser` instead of `MangaParser`

Reply:
The pagesize is only difference when filter with tags, not search query, so `PagedMangaParser` won't work here